### PR TITLE
HOCS-2310 Running custom reports takes too long

### PR DESCRIPTION
**Pull request by agraham19**
* Add refresh endpoint

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/hocs/audit/export/DataExportResource.java [hunk 1]
```diff
@@ -164,6 +165,13 @@ void getCustomDataExport(HttpServletResponse response,
 
     }
 
+    @PostMapping(value = "/export/custom/{viewName}/refresh")
+    public @ResponseBody void refreshMaterialisedView(HttpServletResponse response, @PathVariable("viewName") final String viewName) {
+        int result = customExportService.refreshMaterialisedView(viewName);
+        System.out.println(result);
+        response.setStatus(200);
```
Comment (zuzak) [inline src/main/java/uk/gov/digital/ho/hocs/audit/export/DataExportResource.java:172]: Make sure this only sends 200 if it works. If it fails and the cronjob needs to retry later, it needs to send a 5xx error code.

**Comment by zuzak** [comment]
Switching to draft to prevent accidental merge.

File: src/main/java/uk/gov/digital/ho/hocs/audit/export/DataExportResource.java (hunk 1 above)
Comment (agraham19) [reply src/main/java/uk/gov/digital/ho/hocs/audit/export/DataExportResource.java]: I've changed this and got rid of the uncaught exception.

Comment (Paulcr-ho) [reply src/main/java/uk/gov/digital/ho/hocs/audit/export/DataExportResource.java]: This has been wrapped correctly and returns a 500 on error.

**Review by Paulcr-ho — APPROVED** [review]

**Pull request merged**